python/llm/example/CPU/Applications/autogen/README.md

```diff
@@ -0,0 +1,236 @@
+## Running AutoGen Agent Chat with BigDL-LLM on Local Models
+In this example, we use BigDL adapted FastChat to run [AutoGen](https://microsoft.github.io/autogen/) agent chat with 
+local large language models.
+
+### 1. Prepare Environment
+```bash
+# create autogen running directory
+mkdir autogen
+cd autogen
+
+# create respective conda environment
+conda create -n autogen python=3.9
+conda activate autogen
+
+# install fastchat-adapted bigdl-llm
+pip install --pre --upgrade bigdl-llm[serving]==2.5.0b20240110
+
+# clone the BigDL in the autogen folder
+git clone https://github.com/intel-analytics/BigDL.git
+
+# install recommend transformers version
+pip install transformers==4.36.2
+
+# install necessary dependencies
+pip install chromadb==0.4.22
+```
+
+
+### 2. Setup FastChat and AutoGen Environment
+```bash
+# clone the FastChat in the autogen folder
+git clone https://github.com/lm-sys/FastChat.git FastChat # clone the FastChat
+cd FastChat
+pip3 install --upgrade pip  # enable PEP 660 support
+pip3 install -e ".[model_worker,webui]"
+
+# setup AutoGen environment
+pip install pyautogen==0.2.7
+```
+
+### 3. Build FastChat OpenAI-Compatible RESTful API
+Open 3 terminals
+
+**Terminal 1: Launch the controller**
+
+```bash
+# activate conda environment
+conda activate autogen
+
+# go to the cloned FastChat folder in autogen folder
+cd autogen/FastChat
+
+python -m fastchat.serve.controller
+```
+
+**Terminal 2: Launch the workers**
+
+```bash
+# activate conda environment
+conda activate autogen
+
+# go to the created autogen folder
+cd autogen
+
+# ensure the connection between controller and worker
+export no_proxy='localhost'
+
+# load the local model with cpu with your downloaded model
+python -m bigdl.llm.serving.model_worker --model-path YOUR_MODEL_PATH --device cpu
+```
+
+**Terminal 3: Launch the server**
+
+```bash
+# activate conda environment
+conda activate autogen
+
+# go to the cloned FastChat folder in autogen folder
+cd autogen/FastChat
+
+python -m fastchat.serve.openai_api_server --host localhost --port 8000
+```
+
+### 4. Run Example
+Open another terminal
+
+```bash
+# activate conda environment
+conda activate autogen
+
+# go to the cloned BigDL example folder in autogen folder
+cd autogen/BigDL/python/llm/example/CPU/Applications/autogen
+
+# ensure the connection between controller and worker
+export no_proxy='localhost'
+
+# run the autogen example
+python teachability_new_knowledge.py
+```
+
+
+## [Important] Change the model name
+When you download the model, please change the model folder to `bigdl` for the usage of bigdl adapted FastChat. For example, after download the `Mistral-7B-Instruct-v0.2`
+model and the path is `MODEL_PATH/Mistral-7B-Instruct-v0.2`, please change it to `MODEL_PATH/bigdl`.
```

Comment:
put this important note to step 3 above and show a concreate example. e.g. assume your model is downloaded to "autogen/model/Mistral-7B-Instruct-v0.2", you should change, ... and run the command as 
```bash 
python -m bigdl.llm.serving.model_worker --model-path ... --device cpu
```

Reply:
why we need to change the model folder name? it looks wierd

Reply:
Because in [FastChat](https://github.com/lm-sys/FastChat/blob/main/fastchat/model/model_adapter.py#L2238-L2251) there are a lot of registered adapters that will be called regarding the model name to load the model. For example, we use the `Mistral-7B-Instruct-v0.2` model name and the [mistral adapter](https://github.com/lm-sys/FastChat/blob/main/fastchat/model/model_adapter.py#L1482) will be used. The [BigDL adapter](https://github.com/intel-analytics/BigDL/blob/2d05dbccc6e7813a3df76ff4c6aae2b0ff28f4f2/python/llm/src/bigdl/llm/serving/bigdl_llm_model.py#L281) is registered at the end of the list so we have to rename the model to solely `bigdl` to avoid calling other adapters.